app/controllers/foreman_ansible/api/v2/hosts_controller_extensions.rb

```diff
@@ -36,13 +41,65 @@ def multiple_play_roles
 
             render_message @result
           end
+
+          api :GET, '/hosts/:id/list_ansible_roles',
+              N_('Lists assigned Ansible roles')
+          param :id, :identifier, :required => true
+
+          def list_ansible_roles
+            @result = {
+              :all_ansible_roles => @host.all_ansible_roles,
+              :ansible_roles => @host.ansible_roles,
+              :inherited_ansible_roles => @host.inherited_ansible_roles
+            }
+
+            render_message @result
+          end
+
+          api :POST, '/hosts/:id/ansible_roles',
+              N_('Assigns Ansible roles to a host')
+          param :id, :identifier, :required => true
+          param :roles, Array, :required => true
+
+          def ansible_roles
+            # we do not want to store inherited roles twice
+            @host.ansible_roles = @roles - @host.inherited_ansible_roles
+
+            @result = {
+              :roles => @roles,
+              :host => @host
+            }
+
+            render_message @result
+          end
         end
 
         private
 
+        def find_ansible_roles
```

Comment:
Mind to extract to a shared concern with https://github.com/theforeman/foreman_ansible/pull/81/files#diff-e3a172920758b4851cabcb23ced54a22R86  ?